core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -750,6 +750,7 @@ class OpenMPTargetExec {
                              int64_t thread_local_bytes, int64_t league_size);
 
   static void* m_scratch_ptr;
+  static std::mutex m_mutex_scratch_ptr;
```

Comment:
I just opted for consistency with what we currently have. In the end, probably none of the members here should be `static`, and changing even a single one to be non-`static` requires more changes since an execution space doesn't even have an `OpenMPTargetExec` member. Thus, I would prefer to make those changes in a separate pull request.

Reply:
OK

Reply:
See #6594.